pkg/registry/apis/datasource/sub_resource.go

```diff
@@ -66,12 +67,21 @@ func (r *subResourceREST) Connect(ctx context.Context, name string, opts runtime
 			return
 		}
 
-		path := req.URL.Path[idx+len("/resource"):]
+		clonedReq := req.Clone(req.Context())
```

Comment:
just curious, why the clone?

Reply:
I copied from
https://github.com/grafana/grafana/blob/169be9e40170b15b2561d7e60c00be65ff6f82ee/pkg/api/plugin_resource.go#L90-L100

Reply:
Example if there's some middleware running after so we don't want to change the original request